core/filterconstraint.cpp

```diff
@@ -0,0 +1,939 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "filterconstraint.h"
+#include "dive.h"
+#include "divesite.h"
+#include "errorhelper.h"
+#include "gettextfromc.h"
+#include "qthelper.h"
+#include "tag.h"
+#include "trip.h"
+#include "subsurface-time.h"
+#include <QDateTime>
+
+// We use the units enum only internally.
+// Therefore define it here, not in the header file.
+enum filter_constraint_units {
+	FILTER_CONSTRAINT_NO_UNIT = 0,
+	FILTER_CONSTRAINT_LENGTH_UNIT,
+	FILTER_CONSTRAINT_TEMPERATURE_UNIT,
+	FILTER_CONSTRAINT_WEIGHT_UNIT,
+	FILTER_CONSTRAINT_VOLUMETRIC_FLOW_UNIT
+};
+
+static struct type_description {
+	filter_constraint_type type;
+	const char *s;			// untranslated string that, which will be written to the log
```

Comment:
I would call this `token` or something and point out in the comment that it's supposed to have no space in it

Reply:
done.